source/frontend/src/features/mapSideBar/acquisition/common/GenerateForm/hooks/useGenerateAgreement.ts

```diff
@@ -31,9 +33,11 @@ export const useGenerateAgreement = () => {
     const provincialSolicitor = file.acquisitionTeam?.find(
       team => team.personProfileTypeCode === 'MOTILAWYER',
     );
-    const ownerSolicitor = file.acquisitionFileOwnerSolicitors?.length
-      ? file.acquisitionFileOwnerSolicitors[0]
-      : undefined;
+    const solicitors = file.acquisitionFileInterestHolders
+      ?.filter(x => x.interestHolderType?.id === InterestHolderType.OWNER_SOLICITOR)
+      .filter((x): x is Api_InterestHolder => !!x);
+    const ownerSolicitor =
+      solicitors !== undefined && solicitors.length > 0 ? solicitors[0] : undefined;
```

Comment:
in **useGenerateExpropriationForm1** the following pattern was used. Would lean towards using the same here to keep consistency:

```tsx
    const ownerSolicitor = file.acquisitionFileInterestHolders?.find(
      x => x.interestHolderType?.id === InterestHolderType.OWNER_SOLICITOR,
    );
```

Reply:
is doing the same, the difference is that the second one has potentially many interest holders

Reply:
The 2nd one still only pick the first one it finds- ie "solicitors[0]" so to me that is doing the same as "find" which returns the first it finds. But not a big deal. We can align that later